frameworks/AutoGluon/exec.py

```diff
@@ -64,18 +67,30 @@ def run(dataset, config):
             **training_params
         )
 
-    test_data = TabularDataset(test_path)
     # Persist model in memory that is going to be predicting to get correct inference latency
     predictor.persist_models('best')
 
+    def inference_time_classification(data: Union[str, pd.DataFrame]):
+        return None, predictor.predict_proba(data, as_multiclass=True)
+
+    def inference_time_regression(data: Union[str, pd.DataFrame]):
+        return predictor.predict(data, as_pandas=False), None
+
+    infer = inference_time_classification if is_classification else inference_time_regression
+    inference_times = {}
+    if config.measure_inference_time:
+        inference_times["file"] = measure_inference_times(infer, dataset.inference_subsample_files)
+        test_data = pd.read_parquet(dataset.test.path)
+        inference_times["df"] = measure_inference_times(
```

Comment:
(Suggestion, feel free to create a separate issue as this is mostly a QoL/robustness comment)

We should ensure `inference_times` is the same type structure for all frameworks. Currently AutoGluon has an additional layer of dictionary compared to `GAMA`.

I think `GAMA` should be updated to be `inference_times['file'] = ...`

We may want to have type checking asserts as part of the `return result(..., inference_times=inference_times)` logic that is common to all frameworks.

Something like:

```
if inference_times is not None:
    assert isinstance(inference_times, dict)
    if len(inference_times) > 0:
        for k, v in inference_times:
            assert isinstance(k, str)
            assert isinstance(v, dict)
            for batch_size, infer_time_list in v.items():
                assert isinstance(batch_size, int)
                assert isinstance(infer_time_list, list)
                for infer_time in infer_time_list:
                     assert isinstance(infer_time, float)
                     assert infer_time >= 0
```

Reply:
Yes, I deliberately only updated `AutoGluon` because I felt updating one framework was enough to get an idea for feedback. If all seems well (enough), I will update the other frameworks.

I am not sure about the `assert` type checking. I hope to refactor amlb in the future to contain more type annotations, use mypy, etc. which should catch most of it.